File: scala/src/main/scala/com/spoonacular/client/model/RecipesParseIngredientsNutritionNutrients.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import java.math.BigDecimal

/**
 * 
 * @param name 
 * @param amount 
 * @param unit 
 * @param percentOfDailyNeeds 
 */
case class RecipesParseIngredientsNutritionNutrients(name: String,
                amount: BigDecimal,
                unit: String,
                percentOfDailyNeeds: BigDecimal
                )

object RecipesParseIngredientsNutritionNutrients {
    /**
     * Creates the codec for converting RecipesParseIngredientsNutritionNutrients from and to JSON.
     */
    implicit val decoder: Decoder[RecipesParseIngredientsNutritionNutrients] = deriveDecoder
    implicit val encoder: ObjectEncoder[RecipesParseIngredientsNutritionNutrients] = deriveEncoder
}

File: dart/com/spoonacular/client/model/recipes_parse_ingredients_nutrition_nutrients.dart
part of openapi.api;

class RecipesParseIngredientsNutritionNutrients {
  
  String name = null;
  
  num amount = null;
  
  String unit = null;
  
  num percentOfDailyNeeds = null;
  RecipesParseIngredientsNutritionNutrients();

  @override
  String toString() {
    return 'RecipesParseIngredientsNutritionNutrients[name=$name, amount=$amount, unit=$unit, percentOfDailyNeeds=$percentOfDailyNeeds, ]';
  }

  RecipesParseIngredientsNutritionNutrients.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['name'] == null) {
      name = null;
    } else {
          name = json['name'];
    }
    if (json['amount'] == null) {
      amount = null;
    } else {
          amount = json['amount'];
    }
    if (json['unit'] == null) {
      unit = null;
    } else {
          unit = json['unit'];
    }
    if (json['percentOfDailyNeeds'] == null) {
      percentOfDailyNeeds = null;
    } else {
          percentOfDailyNeeds = json['percentOfDailyNeeds'];
    }
  }

  Map<String, dynamic> toJson() {
    Map <String, dynamic> json = {};
    if (name != null)
      json['name'] = name;
    if (amount != null)
      json['amount'] = amount;
    if (unit != null)
      json['unit'] = unit;
    if (percentOfDailyNeeds != null)
      json['percentOfDailyNeeds'] = percentOfDailyNeeds;
    return json;
  }

  static List<RecipesParseIngredientsNutritionNutrients> listFromJson(List<dynamic> json) {
    return json == null ? new List<RecipesParseIngredientsNutritionNutrients>() : json.map((value) => new RecipesParseIngredientsNutritionNutrients.fromJson(value)).toList();
  }

  static Map<String, RecipesParseIngredientsNutritionNutrients> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, RecipesParseIngredientsNutritionNutrients>();
    if (json != null && json.isNotEmpty) {
      json.forEach((String key, dynamic value) => map[key] = new RecipesParseIngredientsNutritionNutrients.fromJson(value));
    }
    return map;
  }
}

